pydarn/plotting/iq.py

```diff
@@ -0,0 +1,418 @@
+# Copyright (C) 2023 SuperDARN Canada, University of Saskwatchewan
+# Author: Carley Martin
+#
+# Modifications:
+#
+# Disclaimer:
+# pyDARN is under the LGPL v3 license found in the root directory LICENSE.md
+# Everyone is permitted to copy and distribute verbatim copies of this license
+# document, but changing it is not allowed.
+#
+# This version of the GNU Lesser General Public License incorporates the terms
+# and conditions of version 3 of the GNU General Public License,
+# supplemented by the additional permissions listed below.
+#
+
+"""
+IQ plots
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import warnings
+
+from datetime import datetime
+from typing import List
+
+from pydarn import (time2datetime, standard_warning_format, find_record, RTP)
+
+warnings.formatwarning = standard_warning_format
+
+
+class IQ():
+    """
+    Time-series and specialised plots for IQ data
+
+    Class pattern design: Builder
+    This class inherits matplotlib.pyplot to inherit plotting features as well
+    build off their builder design pattern.
+
+    Notes
+    -----
+    This class inherits from matplotlib to generate the figures
+
+    Methods
+    -------
+    plot_time_series
+    plot_iq_sequence
+    plot_iq_record
+    plot_iq_overview
+    """
+
+    def __str__(self):
+        return "This class is static class that provides"\
+                " the following methods: \n"\
+                "   - plot_time_series()\n"\
+                "   - plot_iq_sequence()\n"\
+                "   - plot_iq_record()\n"\
+                "   - plot_iq_overview()\n"
+
+    @classmethod
```

Comment:
I think all the classmethods could be changed to staticmethods. The only class variable is not a true class variable, as it depends on the data passed in to the method and is not an inherent piece of data for the class itself.